src/dataclasses/types.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef _COVIDSIM_DATASTRUCTURES_TYPES_HPP
+#define _COVIDSIM_DATASTRUCTURES_TYPES_HPP
+
+#include <memory>
+
+namespace seir
+{
+    class Population;
+    typedef std::shared_ptr<Population> PopulationPtr;
+
+    class Cell;
+    typedef std::shared_ptr<Cell> CellPtr;
+
+    class Microcell;
+    typedef std::shared_ptr<Microcell> MicrocellPtr;
+
+    class Person;
+    typedef std::shared_ptr<Person> PersonPtr;
+
+    class Place;
+    typedef std::shared_ptr<Place> PlacePtr;
+
+    class Factory;
+
+    enum class InfectionStatus
+    {
+        Susceptible,
+        Exposed,
+        InfectASympt,
+        InfectMild,
+        InfectGP,
+        InfectHosp,
+        InfectICU,
+        InfectICURecov,
+        Recovered,
+        Dead
```

Comment:
Supposing we want to use the software to do a comparison of alternative within-host dynamics which use different categories---would we just have to declare another enum? How will this work with m_status in the Person class?

Reply:
Right now, would have to modify this enum, but it would be easy to modify to allow passing in different enums. (Probably using templating)